icepyx/core/urls.py

```diff
@@ -0,0 +1,10 @@
+from typing import Final
+
+CMR_BASE_URL: Final = "https://cmr.earthdata.nasa.gov"
+GRANULE_SEARCH_BASE_URL: Final = f"{CMR_BASE_URL}/search/granules"
+COLLECTION_SEARCH_BASE_URL: Final = f"{CMR_BASE_URL}/search/collections.json"
+
+EGI_BASE_URL: Final = "https://n5eil02u.ecs.nsidc.org/egi"
+ORDER_BASE_URL: Final = f"{EGI_BASE_URL}/request"
+
+DOWNLOAD_BASE_URL: Final = "https://n5eil02u.ecs.nsidc.org/esir"
```

Comment:
Having just been working on incorporating GEDI into icepyx, I'm noting that these urls are NSIDC specific even though the names are more generic. It's not a dealbreaker, but something we'll have to confront in the near future. 

More generally, I didn't see any immediate analog in earthaccess, but moving towards directly using any urls that are easily surfaceable there is saving us work down the line.